File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.dart
class Solution {
  int maxProfit(List<int> prices) {
    int len = prices.length;
    int l=0,r=1;
    int tempMax = 0;
    // 3, 8, 1, 10,9
    while (r<len){
      if (prices[l]<prices[r]) {
        tempMax = max(tempMax, prices[r] - prices[l]); //6,
      }else{
        l=r;
      }
    r+=1;

    }
    return tempMax;
  }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        #low and high concept
        #l=0
        #[1,4,2]

        l=prices[0] #1
        high=0
        profit = 0
        for i in range(1,len(prices)):
            high = prices[i]    #4
            temp = high - l
            if temp > profit:
                profit = temp
            if prices[i]<l:
                l = prices[i]
        return profit



        
            